taxtastic/subcommands/taxids.py

```diff
@@ -54,7 +55,7 @@ def get_children(engine, parent_ids, rank='species', schema=None):
         for r in rows:
             if r['rank'] == rank and 'sp.' not in r['tax_name']:
                 species.append(r)
-        others = [r for r in rows if r['rank'] not in (rank, 'no_rank')]
+        others = [r for r in rows if r['rank'] not in [rank] + RANK_LOOPS]
```

Comment:
This loop runs many times, so it's probably worth using a set to test for membership:

```python
others = [r for r in rows if r['rank'] not in set([rank] + RANK_LOOPS)]
```